### ISSUE TITLE
CI issue: data race in publishing jobs for `-dev` packages

### ISSUE THREAD

**Issue by leogr** [falcosecurity/falco#1876]
**Describe the bug**

When multiple PRs got merged at the same time, then the CircleCI may run multiple workflows in parallel.
In such a situation, a data race occurs while publishing `-dev` packages to https://download.falco.org/?prefix=packages/
Since the publishing operation also updates RPM and DEB repository metadata, the process may result in corrupted repositories.

This problem happened with #1871 and #1872

However, once another PR gets merged, repository metadata is computed again and the problem disappears.

Only `-dev` packages are affected. Stable packages are not affected because we just tag a release at a time.

**How to reproduce it**

No way to reproduce it locally.

**Expected behaviour**

**Additional context**

I have opened this PR mostly for the record. Likely, we won't fix it.
Final users are not affected by this problem in any way.

**Comment by leogr** [comment]
/remove-lifecycle stale

/help

Just FYI @FedeDP @jasondellaluce

**Comment by leogr** [comment]
/remove-lifecycle rotten

This can still happen.

cc @FedeDP @Andreagit97 pls take this into account if you have in mind to renovate a bit our CI :smile_cat:

**Comment by FedeDP** [comment]
/remove-lifecycle stale

I think the issue might be fixed when porting Ci to GHA and enforcing a correct concurrency in the master workflow.
